custom_components/hacs/base.py

```diff
@@ -467,7 +468,7 @@ def _write_file():
             self.log.error("Could not write data to %s - %s", file_path, error)
             return False
 
-        return os.path.exists(file_path)
+        return await async_exists(self.hass, file_path)
```

Comment:
This check should be part of the `_write_file` function, that should be fixed in a separate PR though.